calculator.c

```diff
@@ -0,0 +1,73 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <ctype.h>
+#include <string.h>
+
+int precedence(char op) {
+    return (op == '*' || op == '/') ? 2 : (op == '+' || op == '-') ? 1 : 0;
+}
+
+int perform(int a, int b, char op, int *error) {
+    if (op == '/' && b == 0) {
+        *error = 1; 
+        return 0;
+    }
+    switch (op) {
+    case '+': return a + b;
+    case '-': return a - b;
+    case '*': return a * b;
+    case '/': return a / b;
+    default: return 0; 
+}
+
+}
+int evaluate(const char *exp,int *result){
+    int values[100],top=-1,error=0;
+    char operators[100] ,opTop=-1;
+    for(int i=0;exp[i];i++){
+        if(isspace(exp[i])) continue;
```

Comment:
Instead of using built-in functions like strchr(), isdigit() and isspace(), Can you create your own definition of these functions and remove unnecessary headerfile?

Reply:
Sure I will make my own function definitions and use them